proposals/handler.go

```diff
@@ -562,6 +557,10 @@ func (h *Handler) getActiveSetWeight(ctx context.Context, id types.Hash32) (uint
 			}
 			return totalWeight, nil
 		}
+	} else {
+		// mark calculation as running
+		h.pendingWeightCalc[id] = nil
+		h.weightCalcLock.Unlock()
 	}
```

Comment:
I think the `else` is unnecessary because the code under `if` always returns.
```suggestion
	}
	// mark calculation as running
	h.pendingWeightCalc[id] = nil
	h.weightCalcLock.Unlock()
```

Reply:
Yes, you're correct, sorry
Fixed